src/mlpack/tests/ann_layer_test.cpp

```diff
@@ -2931,4 +2931,88 @@ BOOST_AUTO_TEST_CASE(ConvolutionLayerPaddingTest)
   module2.Backward(std::move(input), std::move(output), std::move(delta));
 }
 
+/**
+ * Test that the padding options are working correctly in MaxPooling layer.
+ */
+BOOST_AUTO_TEST_CASE(MaxPoolingLayerPaddingTest)
+{
+  arma::mat output, input, delta;
+
+  // Check valid padding option.
+  MaxPooling<> module1(4, 4, 2, 2, true, std::tuple<size_t, size_t>(1, 1),
+      std::tuple<size_t, size_t>(1, 1), "valid");
+
+  // Test the Forward function.
+  input = arma::linspace<arma::colvec>(0, 24, 25);
+  module1.InputWidth() = 5;
+  module1.InputHeight() = 5;
+  module1.Forward(std::move(input), std::move(output));
+
+  BOOST_REQUIRE_EQUAL(arma::accu(output), 18);
+  BOOST_REQUIRE_EQUAL(output.n_rows, 1);
+  BOOST_REQUIRE_EQUAL(output.n_cols, 1);
+
+  // Test the Backward function.
+  module1.Backward(std::move(input), std::move(output), std::move(delta));
+
+  // Check same padding option.
+  MaxPooling<> module2(4, 4, 2, 2, true, std::tuple<size_t, size_t>(0, 0),
+      std::tuple<size_t, size_t>(0, 0), "same");
+
+  // Test the forward function.
+  input = arma::linspace<arma::colvec>(0, 24, 25);
+  module2.InputWidth() = 5;
+  module2.InputHeight() = 5;
+  module2.Forward(std::move(input), std::move(output));
+
+  BOOST_REQUIRE_EQUAL(arma::accu(output), 72);
+  BOOST_REQUIRE_EQUAL(output.n_rows, 4);
+  BOOST_REQUIRE_EQUAL(output.n_cols, 1);
+
+  // Test the backward function.
+  module2.Backward(std::move(input), std::move(output), std::move(delta));
+}
+
+/**
+ * Test that the padding options are working correctly in MeanPooling layer.
+ */
+BOOST_AUTO_TEST_CASE(MeanPoolingLayerPaddingTest)
+{
+  arma::mat output, input, delta;
+
```

Comment:
Hi @tejasvi, Since there was no padding in pooling layers you need to add the following tests as well:
1. Padded output with un-equal padding.
2. Padded with equal padding.
3. Backward test for the same.
4. Test for same padding with unequal padding.
5. Test for same padding with equal padding.

Reply:
The convolution layer is [lacking](https://github.com/mlpack/mlpack/blob/master/src/mlpack/tests/ann_layer_test.cpp#L2903) those as well. That of course is not an excuse to not include them here. I'll add them as well.

Reply:
I have included them in my PR.

Reply:
> Test for same padding with unequal padding.
Test for same padding with equal padding.

Paddings for same are calculated by formula unless you are suggesting to test with uneven input size or stride.

Reply:
I know, you have to add tests such that after InitializeSamePadding calculates padding they shouldn't be equal. (Set inputs s.t padding is uneven).